source/common/redis/command_splitter_impl.cc

```diff
@@ -66,50 +73,22 @@ SplitRequestPtr MGETCommandHandler::startRequest(const RespValue& request,
   single_mget.type(RespType::Array);
   single_mget.asArray().swap(values);
 
-  for (uint64_t i = 1; i < request.asArray().size(); i++) {
-    request_handle->pending_requests_.emplace_back(*request_handle, i - 1);
-    SplitRequestImpl::PendingRequest& pending_request = request_handle->pending_requests_.back();
+  for (uint64_t i = 1; i < incoming_request.asArray().size(); i++) {
+    request_ptr->pending_requests_.emplace_back(*request_ptr, i - 1);
+    PendingRequest& pending_request = request_ptr->pending_requests_.back();
 
-    single_mget.asArray()[1].asString() = request.asArray()[i].asString();
-    ENVOY_LOG(debug, "redis: parallel get: '{}'", single_mget.toString());
-    pending_request.handle_ =
-        conn_pool_.makeRequest(request.asArray()[i].asString(), single_mget, pending_request);
+    single_mget.asArray()[1].asString() = incoming_request.asArray()[i].asString();
+    pending_request.handle_ = conn_pool.makeRequest(incoming_request.asArray()[i].asString(),
+                                                    single_mget, pending_request);
     if (!pending_request.handle_) {
       pending_request.onResponse(Utility::makeError("no upstream host"));
     }
   }
 
-  return request_handle->pending_responses_ > 0 ? std::move(request_handle) : nullptr;
-}
-
-MGETCommandHandler::SplitRequestImpl::SplitRequestImpl(SplitCallbacks& callbacks,
-                                                       uint32_t num_responses)
-    : callbacks_(callbacks), pending_responses_(num_responses) {
-  pending_response_.reset(new RespValue());
-  pending_response_->type(RespType::Array);
-  std::vector<RespValue> responses(num_responses);
-  pending_response_->asArray().swap(responses);
-  pending_requests_.reserve(num_responses);
-}
-
-MGETCommandHandler::SplitRequestImpl::~SplitRequestImpl() {
```

Comment:
Same. Retain destructor and all ASSERTS (if there are others that I am missing).